variant/CompoundCookies.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * Input Vector script to help Zap attack compound cookies, i.e. cookies that contain multiple parameters
+ * Format of a Compound cookie is Classic ASP compliant, i.e. of the form:
+ *        <compoundcookie>=<p1name>=<p1value>&<p2name>=<p2value>&...
+ * where parameter names and values must be URI component encoded. Generates parameters in the form:
+ *        <compoundcookie>:<p1name>=<p1value>
+ *        <compoundcookie>:<p2name>=<p2value>
+ *        ...
+ * These compound cookies should be filtered out in Active Scan Exclude Param to stop ZAP attacking these cookies directly.
+ */
+var ScriptVars    = Java.type('org.zaproxy.zap.extension.script.ScriptVars');
+var HtmlParameter = Java.type('org.parosproxy.paros.network.HtmlParameter');
+var COOKIE_TYPE   = org.parosproxy.paros.network.HtmlParameter.Type.cookie;
+
+/* List of compound cookies to target - either burn in list below (i.e. ccList = [ "<compoundcookie1>", "<compoundcookie2>", ... ]; )
+ * or set via 'CompoundCookies' global var as a '&' separated list (i.e. "<compoundcookie1>&<compoundcookie2>&..." ) */
+var ccList = [ ];
+
+function parseParameters(helper, msg)
+{
+  var headers = msg.getRequestHeader();
+  var cookies = headers.getCookieParams();
+  var ix, ei, pi, pl;

```

Comment:
Maybe declare variables individually and make the names more meaningful (so that others have an easier time modifying /adapting it).

Below as well.

Reply:
I have split out declarations and given more meaningful names to variables